[18.0][MIG] stock_dock: Migration to 18.0

Kimkhoi3010 opened the pull request:
I found only 1 commit to port but it was ignored, I added it to blacklist:

- https://github.com/OCA/stock-logistics-transport/pull/161

During the migration of the stock_dock module, I identified two issues in the tms module that needed to be addressed. To resolve these issues, I created two separate commits:
- Commit 1: The `kanban-box` in the tms module was deprecated in Odoo. This commit replaces the `kanban-box` definition with the recommended `card` template to address the warning.
- Commit 2: The `group_tms_uom` field in the `tms` module had the same label as `group_uom` from the `product` module ("Units of Measure"), causing a conflict. This commit updates the label of group_tms_uom to ensure uniqueness.

rousseldenis commented:
/ocabot migration stock_dock

rousseldenis commented:
@Kimkhoi3010 Thanks for this. Could you remove all changes related to other modules?

You should respect flow for migration:

https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-18.0#technical-method-to-migrate-a-module-from-170-to-180-branch

tms changes should go in another PR.

Kimkhoi3010 commented:
> @Kimkhoi3010 Thanks for this. Could you remove all changes related to other modules?
> 
> You should respect flow for migration:
> 
> https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-18.0#technical-method-to-migrate-a-module-from-170-to-180-branch
> 
> tms changes should go in another PR.

Hi @rousseldenis 
I updated by your comment, I have created a PR to fix the issues in tms module: https://github.com/OCA/stock-logistics-transport/pull/164

bizzappdev commented:
I'm unsure about the necessity of the `checklog_odoo.cfg` file. @Kimkhoi3010, could you please explain its inclusion? 
I'm not aware of any new standards introduced for OCA.

Kimkhoi3010 commented:
> I'm unsure about the necessity of the `checklog_odoo.cfg` file. @Kimkhoi3010, could you please explain its inclusion? I'm not aware of any new standards introduced for OCA.

Hi @bizzappdev,
Sorry for my mistake, I did not update carefully.

rousseldenis commented:
> I'm unsure about the necessity of the `checklog_odoo.cfg` file. @Kimkhoi3010, could you please explain its inclusion? I'm not aware of any new standards introduced for OCA.

@Kimkhoi3010 @bizzappdev That's because on void tests repository main branch, no tests are run. And no tests are declared in this one.

Even if adding that line to configuration solves errors, I always recommend to add a basic test.

@Kimkhoi3010 Here, you  can add a test that checks the default warehouse.

bizzappdev commented:
@Kimkhoi3010 Thank you for the clarification.
@rousseldenis, indeed, adding at least some test cases is better. But strangely, right now, no checklog_odoo.cfg still test builds are passed :) 

@Kimkhoi3010 FYI: https://github.com/OCA/management-system/blob/18.0/checklog-odoo.cfg. The file has been added; you might need to take care of other repos you have added them.

rousseldenis commented:
> indeed, adding at least some test cases is better. But strangely, right now, no checklog_odoo.cfg still test builds are passed :)

Because tms module has been merged

TDu commented on stock_dock/views/stock_dock.xml:
```diff
@@ -24,31 +24,25 @@
                             groups="stock.group_stock_multi_warehouses"
                         />
                         <field name="company_id" groups="base.group_multi_company" />
-                        <field
```
This was added during the 16 migration https://github.com/OCA/stock-logistics-transport/commit/f99f3f069e57b4a9dc3164aaac81e4aebb7cd2a4

Probably for a reason ?

Kimkhoi3010 replied:
I sincerely apologize for my misunderstanding. Thank you for pointing it out.

TDu replied:
Oh, no apology needed, thanks for your work :+1:

Kimkhoi3010 commented:
> > I'm unsure about the necessity of the `checklog_odoo.cfg` file. @Kimkhoi3010, could you please explain its inclusion? I'm not aware of any new standards introduced for OCA.
> 
> @Kimkhoi3010 @bizzappdev That's because on void tests repository main branch, no tests are run. And no tests are declared in this one.
> 
> Even if adding that line to configuration solves errors, I always recommend to add a basic test.
> 
> @Kimkhoi3010 Here, you can add a test that checks the default warehouse.

H

> > I'm unsure about the necessity of the `checklog_odoo.cfg` file. @Kimkhoi3010, could you please explain its inclusion? I'm not aware of any new standards introduced for OCA.
> 
> @Kimkhoi3010 @bizzappdev That's because on void tests repository main branch, no tests are run. And no tests are declared in this one.
> 
> Even if adding that line to configuration solves errors, I always recommend to add a basic test.
> 
> @Kimkhoi3010 Here, you can add a test that checks the default warehouse.

Hi @rousseldenis, @bizzappdev 
I have updated according to your comment, I have added a basic test

bizzappdev reviewed (approved):
Code and Functional Testing LGTM

simahawk commented:
@Kimkhoi3010 pls rewrite the last commit as `[imp] stock_dock: add tests`

Kimkhoi3010 commented:
> @Kimkhoi3010 pls rewrite the last commit as `[imp] stock_dock: add tests`

Hi @simahawk,
I updated this PR by your comment. Thanks for your time.

simahawk commented:
> > @Kimkhoi3010 pls rewrite the last commit as `[imp] stock_dock: add tests`
> 
> Hi @simahawk, I updated this PR by your comment. Thanks for your time.

I still don't see the module name, there's a not so useful ` - ` instead :wink:

Kimkhoi3010 commented:
> > > @Kimkhoi3010 pls rewrite the last commit as `[imp] stock_dock: add tests`
> > 
> > 
> > Hi @simahawk, I updated this PR by your comment. Thanks for your time.
> 
> I still don't see the module name, there's a not so useful `-` instead 😉

Thanks for your time.

rousseldenis commented on stock_dock/__manifest__.py:
```diff
@@ -0,0 +1,21 @@
+{
+    "name": "Loading Dock",
+    "summary": "Manage the loading docks of your warehouse.",
+    "version": "18.0.1.0.1",
```
Normally, you should reset this

```suggestion
    "version": "18.0.1.0.0,
```

Kimkhoi3010 replied:
Thanks for your time.

simahawk commented:
/ocabot merge nobumo

simahawk commented:
/ocabot merge nobump
